### ISSUE TITLE
Overzealous with warnings - does not look into "unsafe" blocks?

### ISSUE THREAD

**Issue by mouse07410** [intellij-rust/intellij-rust#5815]
## Environment

* **IntelliJ Rust plugin version:** 0.3.127.3249-201
* **Rust toolchain version:** 1.47.0-nightly (d6953df14 2020-07-25) x86_64-apple-darwin
* **IDE name and version:** CLion 2020.1.3 (CL-201.8743.17)
* **Operating system:** macOS 10.15.6

## Problem description
I'm getting a false positive:

![IntelliJ-Rust_issue](https://user-images.githubusercontent.com/5923577/88593446-929ba100-d02d-11ea-9c99-170753966907.png)

*Yes I know what `#![deny(warnings)]` does, but even the current `clippy` is perfectly OK with this code - so why is this plugin unhappy?*

`Clippy` output:
```
/Users/ur20980/.cargo/bin/cargo clippy --color=always
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s

Process finished with exit code 0
```
## Steps to reproduce

Try the following Rust code:
```rust
#![feature(asm)]
#![deny(warnings)]
#![deny(clippy::all)]

fn main() {
    let z: i32 = add_new(4, 11);
    println!("4 + 11 = {}", z);
}

fn add_new(a: i32, b: i32) -> i32 {
    let c: i32;
    unsafe {
        asm!(
            "add {0:e}, {1:e}",
            inlateout (reg) a => c,
            in (reg) b,
        );
    }
    c
}
```

**Comment by Kobzol** [comment]
The plugin currently does not understand the new `asm!` macro and therefore does not consider things like inout variables, therefore it does not see any usage/initialization of `c`. I'll try to implement some basic support, the corresponding issue is here: 
https://github.com/intellij-rust/intellij-rust/issues/1940.

**Comment by mouse07410** [comment]
@Kobzol thank you very much! Don't mean to be a pest - but would you have an idea when this support is likely to show up in a released plugin?

**Comment by Kobzol** [comment]
(Just a disclaimer, I'm not a maintainer of the plugin, I just contribute to it). I don't know :) I'll try to create a PR with some basic support, but it will be a relatively complex feature, so it might take some time (probably a few weeks at least).
